src/core/MIGRATION_EXAMPLES.md

```diff
@@ -1003,4 +1007,178 @@ setup(core: CoreSetup){
     },
   })
 }
-```
\ No newline at end of file
+```
+
+## Elasticsearch client
+
+The new elasticsearch client is a thin wrapper around `@elastic/elasticsearch`'s `Client` class. Even if the API
+is quite close to the legacy client Kibana was previously using, there are some subtle changes to take into account 
+during migration.
+
+[Official documentation](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/index.html) 
+
+### Client API Changes
+
+The most significant changes for the consumers are the following:
+
+- internal / current user client accessors has been renamed and are now properties instead of functions
+  - `callAsInternalUser('ping')` -> `asInternalUser.ping()`
+  - `callAsCurrentUser('ping')` -> `asCurrentUser.ping()`
+
+- the API now reflects the `Client`'s instead of leveraging the string-based endpoint names the `LegacyAPICaller` was using
+
+```ts
+// before
+const body = await client.callAsInternalUser('indices.get', { index: 'id' });
+// after
+const { body } = await client.asInternalUser.indices.get({ index: 'id' });
+```
+
+- calling any ES endpoint now returns the whole response object instead of only the body payload
+
+```ts
+// before
+const body = await legacyClient.callAsInternalUser('get', { id: 'id' });
+// after
+const { body } = await client.asInternalUser.get({ id: 'id' });
+```
+
+Note that more information from the ES response are available:
+
+```ts
+const {
+  body,        // response payload
+  statusCode,  // http status code of the response
+  headers,     // response headers
+  warnings,    // warnings returned from ES
+  meta         // meta information about the request, such as request parameters, number of attempts and so on
+} = await client.asInternalUser.get({ id: 'id' });
+```
+
+- all API methods are now generic to allow specifying the response body type
+
+```ts
+// before
+const body: GetResponse = await legacyClient.callAsInternalUser('get', { id: 'id' });
+```
+
+```ts
+// body is of type `GetResponse`
+const { body } = await client.asInternalUser.get<GetResponse>({ id: 'id' });
```

Comment:
Should we mention that `GetResponse` should be imported from the `src/core/server`?

Reply:
This is a question we have yet to answer imho. Do we want to maintain and expose the types for the most commonly used ES API responses?

Reply:
we don't want to do this in the long run. However, it is better than not providing any typings at all. Ok, let's skip for now, probably we can have https://github.com/elastic/elasticsearch-js/pull/970 merged soon